homeassistant/components/strava/sensor.py

```diff
@@ -0,0 +1,401 @@
+                               discovery_info=None):
+
+    data = hass.data.get(DOMAIN)
+
+    athlete_id = config.get(CONF_ATHLETE)
+    gear_id = config.get(CONF_GEAR)
+    club_id = config.get(CONF_CLUB)
+    bike_id = config.get(CONF_BIKE)
+    shoe_id = config.get(CONF_SHOE)
+
+    fields = config.get(CONF_FIELDS) or []
+    stats = config.get(CONF_STATS) or []
+    activity = config.get(CONF_ACTIVITY) or []
+
+    sensors = []
+
+    if bike_id:
+        gear_id = 'b{}'.format(bike_id)
+    elif shoe_id:
+        gear_id = 'g{}'.format(shoe_id)
+
+    if athlete_id:
+        if athlete_id == 'me':
+            athlete_id = None
+
+        for field in stats:
+            sensor = StravaAthleteStatsSensor(data, athlete_id, field)
+            sensors.append(sensor)
+
+        for field in activity:
+            sensor = StravaLastActivitySensor(data, athlete_id, field)
+            sensors.append(sensor)
+
+        for field in fields:
+            sensor = StravaAthleteDetailsSensor(data, athlete_id, field)
+            sensors.append(sensor)
+
+    elif gear_id:
+        for field in fields:
+            sensor = StravaGearSensor(data, gear_id, field)
+            sensors.append(sensor)
+
+    elif club_id:
+        for field in fields:
+            sensor = StravaClubSensor(data, club_id, field)
+            sensors.append(sensor)
+
+    add_entities(sensors, True)
+
+
+class StravaSensor(Entity):
+
+    def __init__(self, field):
+        self._state = None
+
+        if '.' in field:
+            comps = field.split('.')
+
+            self._field = comps[0]
+            self._subfield = comps[1]
+        else:
+            self._field = field
+            self._subfield = None
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        from stravalib.model import ActivityTotals
+        from units.quantity import Quantity
+
+        attr = getattr(self._state, self._field, None)
+        if isinstance(attr, ActivityTotals):
+            attr = getattr(attr, self._subfield, None)
+
+        if isinstance(attr, Quantity):
+            return attr.num
+        else:
+            return attr
+
+    @property
+    def unit_of_measurement(self):
+        from stravalib.model import ActivityTotals
+        from units.quantity import Quantity
+
+        attr = getattr(self._state, self._field, None)
+        if isinstance(attr, ActivityTotals):
+            attr = getattr(attr, self._subfield, None)
+
+        if isinstance(attr, Quantity):
+            return str(attr.unit)
+        else:
+            return None
+
+    @property
+    def icon(self):
+        if self._field and self._field in ICON_MAPPING_FIELDS:
+            return ICON_MAPPING_FIELDS[self._field]
+        elif self._subfield and self._subfield in ICON_MAPPING_FIELDS:
+            return ICON_MAPPING_FIELDS[self._subfield]
+        else:
+            return ICON
+
+    @property
+    def available(self):
+        return self._state
```

Comment:
Is that a boolean?

Reply:
Thanks! Fixed now.

I also added some try except blocks around the state updates to reset the `self._state` member.